EarlySamples/InitialProject/InitialProject/Assets/Scripts/Test2/TestPlayerInputManager.cs

```diff
@@ -0,0 +1,107 @@
+    InputAction jumpInputAction;
+    InputAction moveLeftAction;
+    InputAction moveRightAction;
+    InputAction moveUpAction;
+    InputAction moveDownAction;
+
+    public static event EventHandler onLeftImpact; // The event to signal player left collison with wall
+    public static event EventHandler onRightImpact; // The event to signal player right collision with wall
+    public static event EventHandler RunRequest;
+    public static event EventHandler WalkRequest;
+    public static event EventHandler BackwardsRequest;
+    public static event EventHandler StrafeLeftRequest;
+    public static event EventHandler StrafeRightRequest;
+
+    public static event EventHandler moveLeft;
+    public static event EventHandler moveRight;
+    public static event EventHandler moveUp;
+    public static event EventHandler moveDown;
+    public static event EventHandler jump;
+
+    private void Awake()
+    {
+        if (instance != null && instance != this) //eliminate all duplicates of DummyTestGameManager instance
+        {
+            Destroy(gameObject);
+        }
+        else
+        {
+            instance = this;
+            DontDestroyOnLoad(this.gameObject);
+        }
+
+        keyboardActionMap = testActions.FindActionMap("Player");
+        moveLeftAction = keyboardActionMap.FindAction("MoveLeft");
+        moveRightAction = keyboardActionMap.FindAction("MoveRight");
+        moveUpAction = keyboardActionMap.FindAction("MoveUp");
+        moveDownAction = keyboardActionMap.FindAction("MoveDown");
+        jumpInputAction = keyboardActionMap.FindAction("Jump");
+
+        jumpInputAction.performed += HandleJumpInputEvent;
+        moveRightAction.performed += HandleMoveRightEvent;
+        moveLeftAction.performed += HandleMoveLeftEvent;
+        moveUpAction.performed += HandleMoveUpEvent;
+        moveDownAction.performed += HandleMoveDownEvent;
+
+    }
+
+    private void HandleMoveDownEvent(InputAction.CallbackContext obj)
+    {
+        moveDown(this, EventArgs.Empty);
+    }
+
+    private void HandleMoveUpEvent(InputAction.CallbackContext obj)
+    {
+        moveUp(this, EventArgs.Empty);
+    }
+
+    private void HandleMoveLeftEvent(InputAction.CallbackContext obj)
+    {
+        moveLeft(this, EventArgs.Empty);
+    }
+
+    private void HandleMoveRightEvent(InputAction.CallbackContext obj)
+    {
+        moveRight(this, EventArgs.Empty);
+    }
+
+    private void HandleJumpInputEvent(InputAction.CallbackContext obj)
+    {
+        jump(this, EventArgs.Empty);
+    }
+
+    //Handle all action triggers by comparing name, and invoking the appropriate event launch function
+    
+
+    //The function to fire of the leftImpact request
+    private void OnLeftImpactRequest()
+    {
+        if (onLeftImpact != null)
+        {
+            onLeftImpact(this, EventArgs.Empty);
+        }
+    }
+
+    //The function to fire of the rightImpact request
+    private void OnRightImpactRequest()
```

Comment:
Unused?

Reply:
Function moved from this class. During corrections was wondering if the request on impact should come from playerCollisionHandling class? This class handles player movement and doesn't seem cohesive with impacts and collision. In the collision impact class, the onLeftImpact event can be directly accessed and launched. The event has been subscribed to and is being handled in the PlayerMovementHandler(which can also be changed if needed). The checks required for calling the event can be done in the collision class.

Reply:
We can talk about this. Seems like this was just testing.